paperqa/agents/task.py

```diff
@@ -487,3 +497,214 @@ def __len__(self) -> int:
     LitQAv2TaskDataset.__module__,
     LitQAv2TaskDataset.__name__,
 )
+
+
+class LFRQAPairwiseEvalEnv(GradablePaperQAEnvironment):
+    """Environment to evaluate paperqa's vs human's answers on Long Form RAG QA questions."""
+
+    def __init__(
+        self,
+        *args,
+        qid: str | UUID,
+        question: str,
+        human_answer: str,
+        gt_doc_ids: list[str],
+        pairwise_eval_llm: LLMModel | str = CommonLLMNames.GPT_4O.value,
+        evaluation_callback: (
+            Callable[[MultipleChoiceEvaluation | dict], Awaitable] | None
+        ) = None,
+        **kwargs,
+    ):
+        kwargs["query"] = question
+        kwargs["docs"] = Docs()
+        super().__init__(*args, **kwargs)
+
+        self.qid = qid
+        self.question = question
+        self.human_answer = human_answer
+        self.gt_doc_ids = gt_doc_ids
+        self.pairwise_eval_llm = pairwise_eval_llm
+        self._evaluation_callback = evaluation_callback
+
+    def extract_best_answer_index(self, text: str) -> int:
+        match = re.search(r"<rating>(\d+)</rating>", text)
+        return int(match.group(1)) if match else 0
+
+    async def pairwise_evaluation(
+        self, qid: str | UUID, question: str, pqa_answer: str, human_answer: str
+    ) -> dict:
+
+        paper_search_ids = [int(doc.docname) for doc in self.state.docs.docs.values()]
+
+        pairwise_eval_llm = LiteLLMModel(name=self.pairwise_eval_llm)
+        pqa_answer = strip_citations(pqa_answer)
+
+        pqa_answer_index = 1 if random.random() < 0.5 else 2  # noqa: PLR2004
+        data = {
+            "question": question,
+            "answer1": pqa_answer if pqa_answer_index == 1 else human_answer,
+            "answer2": human_answer if pqa_answer_index == 1 else pqa_answer,
+        }
+
+        result = await pairwise_eval_llm.run_prompt(
+            prompt=lfrqa_prompt_template,
+            data=data,
+            system_prompt=lfrqa_system_prompt,
+        )
+
+        best_answer_index = self.extract_best_answer_index(result.text)
+        winner = (
+            "paperqa"
+            if best_answer_index == pqa_answer_index
+            else "human" if best_answer_index != 0 else "tie"
+        )
```

Comment:
```suggestion
        if best_answer_index == pqa_answer_index:
            winner, reward = "paperqa", self._rewards["win"]
        elif best_answer_index != 0:
            winner, reward = "human", self._rewards["lose"]
        else:
            winner, reward = "tie", self._rewards["tie"]
```

Just consolidating logic